File: server/models/database_schema.py
from typing import List, Optional
from pydantic import BaseModel, Field


class Column(BaseModel):
    name: str
    type: str
    primary_key: Optional[bool] = False
    foreign_key: Optional[str] = None


class Table(BaseModel):
    name: str
    columns: List[Column]


class Relationship(BaseModel):
    from_table: str
    to_table: str
    relationship_type: str
    on_column: str


class DatabaseSchema(BaseModel):
    tables: List[Table] = Field(default_factory=list)
    relationships: List[Relationship] = Field(default_factory=list)

    class Config:
        extra = "forbid"

File: client2/src/model/database-schema.ts
export type Column = {
    name: string;
    type: string;
    primary_key?: boolean;
    foreign_key?: string;
};

export type Table = {
    name: string;
    columns: Column[];
};

export type Relationship = {
    from_table: string;
    to_table: string;
    relationship_type: string;
    on_column: string;
};

export type DatabaseSchema = {
    tables: Table[];
    relationships: Relationship[];
};
